test/integration/testsuite/minishift.go

```diff
@@ -350,3 +350,26 @@ func (m *Minishift) minishiftHomeDirectoryShouldntExist(dir string) error {
 
 	return fmt.Errorf("Directory %s exists", dir)
 }
+
+func (m *Minishift) setEnvironmentVariable(variable string, value string) error {
+	os.Setenv(variable, value)
```

Comment:
https://golang.org/pkg/os/#Setenv this return the error if not able to set the variable to env. so you need to handle it properly and below block is not required.

```golang
if os.Getenv(variable) != value {
	return fmt.Errorf("Error in setting environment variable %s=%s in the current shell", variable, value)	
}
```